tests/ui/transmutability/assoc-bound.rs

```diff
@@ -0,0 +1,25 @@
+#![crate_type = "lib"]
+#![feature(transmutability)]
+
+trait A {
+    type AssocA;
+}
+
+trait B {
+    type AssocB: std::mem::TransmuteFrom<()>;
+}
+
+impl<T> B for (T, u8)
+where
+    T: A,
+{
+    type AssocB = T::AssocA; //~ERROR: the trait bound `<T as A>::AssocA: TransmuteFrom<(), Assume { alignment: false, lifetimes: false, safety: false, validity: false }>` is not satisfied [E0277]
```

Comment:
I'm personally fine to punt on making this error message prettier — my top priority is making `TransmuteFrom` work — but happy to try to improve this further if you'd like to block the PR on it.

Reply:
No, it's fine. But yeah, it could be customized to say something like "cannot transmute types with ty/ct params in it".

Reply:
But happy to do that later.